From 9f8cd57fcbc8ec92057a48a20b69179de53779be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 21 Jan 2022 21:46:08 +0100
Subject: [PATCH 26/28] winegstreamer: Use an optional h264parse wg_transform
 element.

Required for Mortal Kombat 11.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=45988
Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=47084
Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=49715
Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=52183
CW-Bug-Id: #16839
CW-Bug-Id: #18678
CW-Bug-Id: #19362
---
 dlls/winegstreamer/wg_transform.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/dlls/winegstreamer/wg_transform.c b/dlls/winegstreamer/wg_transform.c
index 11111111111..11111111111 100644
--- a/dlls/winegstreamer/wg_transform.c
+++ b/dlls/winegstreamer/wg_transform.c
@@ -366,6 +366,17 @@ NTSTATUS wg_transform_create(void *args)
     transform->container = gst_bin_new("wg_transform");
     assert(transform->container);
 
+    switch (input_format.encoded_type)
+    {
+    case WG_ENCODED_TYPE_H264:
+        if ((element = create_element("h264parse", "base")) &&
+                !transform_append_element(transform, element, &first, &last))
+            goto failed;
+        break;
+    default:
+        break;
+    }
+
     if (!(element = try_create_transform(src_caps, raw_caps)) ||
             !transform_append_element(transform, element, &first, &last))
         goto failed;
-- 
2.35.1

